### PR TITLE
fix: export default the email provider

### LINKED ISSUE #6497: SvelteKitAuth: No Call signature for Email in @auth/core/providers/email
### Environment

  System:
    OS: Linux 6.0 Pop!_OS 22.04 LTS
    CPU: (8) x64 Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz
    Memory: 2.70 GB / 7.60 GB
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 18.12.1 - /usr/bin/node
    npm: 9.2.0 - /usr/bin/npm
  Browsers:
    Firefox: 108.0

### Reproduction URL

https://github.com/nextauthjs/sveltekit-auth-example

### Describe the issue

his expression is not callable.
  Type 'typeof import(".../node_modules/@auth/core/providers/email")' has no call signatures.ts(2349)

--> build error:
TypeError: __vite_ssr_import_4__.default.default is not a function at /home/.../src/hooks.server.ts:47:17 at async instantiateModule (file:///home/.../node_modules/vite/dist/node/chunks/dep-5e7f419b.js:52224:9)

### How to reproduce

   1. Clone the repo
   2. Install the repo
   3.  npm run dev

### Expected behavior

expected build to complete without errors.

### PR DESCRIPTION
**Pull request by OrJDev**
> _NOTE_:
>
> - It's a good idea to open an issue first to discuss potential changes.
> - Please make sure that you are _NOT_ opening a PR to fix a potential security vulnerability. Instead, please follow the [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md) to disclose the issue to us confidentially.

## 🧢 Checklist

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

## 🎫 Affected issues

Please [scout and link issues](https://github.com/nextauthjs/next-auth/issues) that might be solved by this PR.

Fixes: INSERT_ISSUE_LINK_HERE

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

Closes https://github.com/nextauthjs/next-auth/issues/6497

### CODE REVIEW

File: packages/core/src/providers/email.ts [hunk 1]
```diff
@@ -111,6 +111,8 @@ export function Email(config: EmailConfig): EmailConfig {
   }
 }
 
+export default Email
```
Comment (binaryme) [inline packages/core/src/providers/email.ts:114]: I tried hardcoding this into my node_modules/@auth/core/src/providers/email.ts but the error persist

Comment (OrJDev) [reply packages/core/src/providers/email.ts:114]: ye because it should be email.js not .ts

Comment (binaryme) [reply packages/core/src/providers/email.ts:114]: What else could I be missing? 
I'm sure I added that change to 
`@auth/core/providers/email.js`

Comment (OrJDev) [reply packages/core/src/providers/email.ts:114]: > What else could I be missing? I'm sure I added that change to `@auth/core/providers/email.js`

the issue is that i wasn't default exported so it throws this issue, i tried this using solid-start (vite as-well) and it seemed to throw the exact same issue, after adding this line to email.js this issue disappears

Comment (OrJDev) [reply packages/core/src/providers/email.ts:114]: > 
make sure you add this to email.js and restart the dev server.

**Review by binaryme — APPROVED** [review]

**Review by balazsorban44 — COMMENTED** [review]
Thanks. Although going from default to a named export only on this provider went a bit unnoticed and we should have done it more holistically (for all providers at the same time), my opinion is that we should use as few default exports as possible, and as such, we should rather change this in other providers, and not revert this one.

If for nothing else, but to get auto-completion in IDEs like VSCode.

I would definitely not like to have both a named and a default export though.

What do you think?

**Comment by binaryme** [comment]
I can confirm I can use the login with email when using the named export Email in the providers array

**Comment by OrJDev** [comment]
> Thanks. Although going from default to a named export only on this provider went a bit unnoticed and we should have done it more holistically (for all providers at the same time), my opinion is that we should use as few default exports as possible, and as such, we should rather change this in other providers, and not revert this one.
> 
> 
> 
> If for nothing else, but to get auto-completion in IDEs like VSCode.
> 
> 
> 
> I would definitely not like to have both a named and a default export though.
> 
> 
> 
> What do you think?

I also think there should only be 1 export, 
IDEs like vsCode give you auto completion for both named and default export, so it depends if you want to keep the old nextAuth api reference or move over, this ofc will confuse previous users and there seems to be nothing wrong with it being default exported so maybe we should just stick to default?

**Comment by canastro** [comment]
Don't you have an issue with typescript requiring you to pass "id", "type", "name" and "sendVerificationRequest"? It doesnt seem like the export is only thing that is currently wrong.

**Pull request closed without merging**